skore/src/skore/sklearn/_estimator/report.py

```diff
@@ -100,6 +102,18 @@ def __init__(
         else:  # fit is False
             self._estimator = estimator
 
+        try:
+            self._estimator = copy.deepcopy(self._estimator)
```

Comment:
Here, we need to make sure to only do the deepcopy when `fit == False` (the `else` branch` or in the `try` part of the `"auto"` branch. You can factor out this `try/except` inside a function called `_copy_estimator` since we use it twice.

Reply:
done